src/storage/index/hash_index.cpp

```diff
@@ -131,6 +131,14 @@ void HashIndex<T>::splitSlots(HashIndexHeader& header, slot_id_t numSlotsToSplit
     // So instead buffer new overflow slots here and append them at the end
     std::vector<Slot<T>> newOverflowSlots;
 
+    auto getNextOvfSlot = [&](slot_id_t nextOvfSlotId) {
+        if (nextOvfSlotId >= oSlots->getNumElements()) {
```

Comment:
Better assert here `nextOvfSlotId - oSlots->getNumElements()` is within the range of `newOverflowSlots.size()`